students/JoeNunnelley/session02/series.py

```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env python3
+
+"""
+Lesson 2 : The Series Project
+Author : Joe Nunnelley
+"""
+
+def fibonacci(n):
+    """Return the nth fibonacci number"""
+
+    # sanitize input by ensuring its an integer
+    n = int(n)
+
+    if n > 1:
+        return fibonacci(n - 1) + fibonacci(n - 2)
+    elif n == 1:
+        return 1
+    elif n <= 0:
+        return 0
+
+def lucas(n):
+    """Return nth lucas number"""
+
+    # sanitize input by ensuring its an integer
+    n = int(n)
+
+    if n > 1:
+        return lucas(n - 1) + lucas(n - 2)
+    elif n == 1:
+        return 1
+    elif n <= 0:
+        return 2
+
+def sum_series(n, first=0, second=1):
+    """
+    compute the nth value of a summation series.
+
+    :param n0=0: value of zeroth element in the series
+    :param n1=1: value of first element in the series
+
+    This function should generalize the fibonacci() and the lucas(),
+    so that this function works for any first two numbers for a sum series.
+    Once generalized that way, sum_series(n, 0, 1) should be equivalent to fibonacci(n).
+    And sum_series(n, 2, 1) should be equivalent to lucas(n).
+    """
+
+    # sanitize input by ensuring its an integer
+    n = int(n)
+
+    if not first:
+        """Return the fibonacci number for the value provided"""
+        if n > 1:
+            return fibonacci(n - 1) + fibonacci(n - 2)
+        if n == 1:
+            return second
+        elif n <= 0:
+            return first
+    else:
```

Comment:
Hey Joe, it seems I spoke too early for this submission. You handle the Fibonacci and Lucas cases but you will want to have a generalized solution that can handle any n0 and n1. Look into a solution that will satisfy any choice of n0 and n1 (this way it will also handle fibonacci and lucas). Please submit another PR when you have the solution.

Let me know if you need me to clarify anything, thanks!